### PR TITLE
refactor(github): Change github-related field types for repo cache

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Ref: #27641

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/cache/repository/types.ts [hunk 1]
```diff
@@ -144,7 +144,13 @@ export interface RepoCacheData {
     gitea?: {
       pullRequestsCache?: GiteaPrCacheData;
```
Comment (viceice) [inline lib/util/cache/repository/types.ts:145]: we should also use unknown here and on other places to not have references to platform.

should of cause be done in separate PR

File: lib/modules/platform/github/pr.ts [hunk 2]
```diff
@@ -12,7 +12,6 @@ function getPrApiCache(): ApiCache<GhPr> {
   const repoCache = getCache();
   repoCache.platform ??= {};
   repoCache.platform.github ??= {};
-  delete repoCache.platform.github.prCache;
```
Comment (viceice) [inline lib/modules/platform/github/pr.ts:15]: why this?

Comment (zharinov) [reply lib/modules/platform/github/pr.ts:15]: This field was last used 1.5 years ago, so hopefully we've migrated all cache instances since then.

**Review by viceice — APPROVED** [review]
